src/components/timeline/TrackContextMenu.js

```diff
@@ -419,14 +436,42 @@ class TimelineTrackContextMenu extends PureComponent<Props> {
     );
   }
 
+  renderIsolateScreenshot() {
+    const { rightClickedTrack, globalTracks, hiddenGlobalTracks } = this.props;
+
+    if (rightClickedTrack === null) {
+      return null;
+    }
+
+    const track = globalTracks[rightClickedTrack.trackIndex];
+    if (track.type !== 'screenshots') {
+      // Only process tracks with a main thread can be isolated.
+      return null;
+    }
+
+    const isDisabled =
+      // Is there only one global track and one screenshot visible?
+      globalTracks.length - hiddenGlobalTracks.size === 2;
```

Comment:
probably this check is not good: you're not checking screenshot tracks here, but all tracks.
You can look at my code suggestion for the action, and use something similar here to define this condition.

Reply:
Hmmm, just spent over an hour on this.
I came over this logic, 
```
findTotalVisibleScreenshots(globalTracks, hiddenGlobalTracks) {
    console.log(hiddenGlobalTracks);
    let screenshotTypeTracks = 0;
    let hiddenScreenshotTypeTracks = 0;
    for (const globalTrack of globalTracks) {
      if (globalTrack.type === 'screenshots') {
        screenshotTypeTracks++;
      }
    }
    for (const hiddenGlobalTrack of hiddenGlobalTracks) {
      console.log(hiddenGlobalTrack);
      if (hiddenGlobalTrack.type === 'screenshots') {
        hiddenScreenshotTypeTracks++;
      }
    }
    console.log(screenshotTypeTracks, hiddenScreenshotTypeTracks);
    return screenshotTypeTracks - hiddenScreenshotTypeTracks === 1;
  }
```

Until I realized that I don't know any way to find out how many screenshots are actually visible. 
I thought hiddenGlobalTracks would give me a count of the screenshots that were hidden and then I could subtract that from the number of GlobalTracks of type screenshots.

I later realized that hiddenGlobalTracks is just a array of numbers, and cannot help in that way.

Coming back to the logic I have already written,
```globalTracks.length - hiddenGlobalTracks.size === 2;```
This also obviously fails in many cases as well. 

>You can look at my code suggestion for the action

Are you talking about code in [this](https://github.com/firefox-devtools/profiler/pull/2232#discussion_r321826018) comment?

If yes, the logic I wrote in this comment above was in accordance with your suggested code, and I'm stuck!

A little help needed here! @julienw 

One way could also be to leave out the '''isDisabled''' all together, as even if the user clicks on "Hide all other screenshots", there won't be any difference.

What do you think?

Any suggestion/hint is appreciated! Thanks